src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -29,28 +29,39 @@ class MessengerPass implements CompilerPassInterface
 {
     use PriorityTaggedServiceTrait;
 
-    private $messageBusService;
-    private $messageHandlerResolverService;
     private $handlerTag;
+    private $busTag;
+    private $senderTag;
+    private $receiverTag;
 
-    public function __construct(string $messageBusService = 'message_bus', string $messageHandlerResolverService = 'messenger.handler_resolver', string $handlerTag = 'messenger.message_handler')
+    public function __construct(string $handlerTag = 'messenger.message_handler', string $busTag = 'messenger.bus', string $senderTag = 'messenger.sender', string $receiverTag = 'messenger.receiver')
     {
-        $this->messageBusService = $messageBusService;
-        $this->messageHandlerResolverService = $messageHandlerResolverService;
         $this->handlerTag = $handlerTag;
+        $this->busTag = $busTag;
+        $this->senderTag = $senderTag;
+        $this->receiverTag = $receiverTag;
     }
 
     /**
      * {@inheritdoc}
      */
     public function process(ContainerBuilder $container)
     {
-        if (!$container->hasDefinition($this->messageBusService)) {
+        if (!$container->hasDefinition('messenger.handler_resolver')) {
             return;
         }
 
-        if (!$container->getParameter('kernel.debug') || !$container->hasAlias('logger')) {
-            $container->removeDefinition('messenger.middleware.debug.logging');
+        $collectorDefinition = $container->getDefinition('messenger.data_collector');
```

Comment:
kind of couples the pass to WebProfiler (basically Framework), I would skip if the definition does not exist

Reply:
Fair enough.